<html>
<head><meta charset="utf-8"><title>Timeout Error re-entry · wasmtime · Zulip Chat Archive</title></head>
<h2>Stream: <a href="https://bytecodealliance.github.io/zulip-archive/stream/217126-wasmtime/index.html">wasmtime</a></h2>
<h3>Topic: <a href="https://bytecodealliance.github.io/zulip-archive/stream/217126-wasmtime/topic/Timeout.20Error.20re-entry.html">Timeout Error re-entry</a></h3>

<hr>

<base href="https://bytecodealliance.zulipchat.com">

<head><link href="https://bytecodealliance.github.io/zulip-archive/style.css" rel="stylesheet"></head>

<a name="442513522"></a>
<h4><a href="https://bytecodealliance.zulipchat.com#narrow/stream/217126-wasmtime/topic/Timeout%20Error%20re-entry/near/442513522" class="zl"><img src="https://bytecodealliance.github.io/zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Ashanti Mutinta <a href="https://bytecodealliance.github.io/zulip-archive/stream/217126-wasmtime/topic/Timeout.20Error.20re-entry.html#442513522">(Jun 04 2024 at 12:04)</a>:</h4>
<p>I have this weird bug in 17.0.3, I have a set a callback that yields until it reaches a max or final value. If this value is reached it returns a trap::interrupt error. Everything seems fine but as soon as a timeout each time I create a new sandbox and invoke a function on it, it seems like it tells me "cannot enter component instance" thought I would ask if anyone has seen anything similar?</p>



<a name="442519805"></a>
<h4><a href="https://bytecodealliance.zulipchat.com#narrow/stream/217126-wasmtime/topic/Timeout%20Error%20re-entry/near/442519805" class="zl"><img src="https://bytecodealliance.github.io/zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Lann Martin <a href="https://bytecodealliance.github.io/zulip-archive/stream/217126-wasmtime/topic/Timeout.20Error.20re-entry.html#442519805">(Jun 04 2024 at 12:37)</a>:</h4>
<p>After a component instance traps it is no longer usable</p>



<a name="442519973"></a>
<h4><a href="https://bytecodealliance.zulipchat.com#narrow/stream/217126-wasmtime/topic/Timeout%20Error%20re-entry/near/442519973" class="zl"><img src="https://bytecodealliance.github.io/zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Ashanti Mutinta <a href="https://bytecodealliance.github.io/zulip-archive/stream/217126-wasmtime/topic/Timeout.20Error.20re-entry.html#442519973">(Jun 04 2024 at 12:38)</a>:</h4>
<p>Even if it is a new sandbox? I guess I could just change the type of error returned</p>



<a name="442520395"></a>
<h4><a href="https://bytecodealliance.zulipchat.com#narrow/stream/217126-wasmtime/topic/Timeout%20Error%20re-entry/near/442520395" class="zl"><img src="https://bytecodealliance.github.io/zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Lann Martin <a href="https://bytecodealliance.github.io/zulip-archive/stream/217126-wasmtime/topic/Timeout.20Error.20re-entry.html#442520395">(Jun 04 2024 at 12:40)</a>:</h4>
<p>Could you expand on what you mean by "new sandbox"? The key type here in wasmtime would be the <code>Instance</code></p>



<a name="442520615"></a>
<h4><a href="https://bytecodealliance.zulipchat.com#narrow/stream/217126-wasmtime/topic/Timeout%20Error%20re-entry/near/442520615" class="zl"><img src="https://bytecodealliance.github.io/zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Ashanti Mutinta <a href="https://bytecodealliance.github.io/zulip-archive/stream/217126-wasmtime/topic/Timeout.20Error.20re-entry.html#442520615">(Jun 04 2024 at 12:41)</a>:</h4>
<p>Oh yes sorry about that. A new instance being built from a preinit-module</p>



<a name="442520961"></a>
<h4><a href="https://bytecodealliance.zulipchat.com#narrow/stream/217126-wasmtime/topic/Timeout%20Error%20re-entry/near/442520961" class="zl"><img src="https://bytecodealliance.github.io/zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Lann Martin <a href="https://bytecodealliance.github.io/zulip-archive/stream/217126-wasmtime/topic/Timeout.20Error.20re-entry.html#442520961">(Jun 04 2024 at 12:43)</a>:</h4>
<p>That should be fine. The error you are seeing means that the component has been entered and not "cleanly" exited; trapping would be the most obvious reason for that in this case <span aria-label="shrug" class="emoji emoji-1f937" role="img" title="shrug">:shrug:</span></p>



<a name="442521098"></a>
<h4><a href="https://bytecodealliance.zulipchat.com#narrow/stream/217126-wasmtime/topic/Timeout%20Error%20re-entry/near/442521098" class="zl"><img src="https://bytecodealliance.github.io/zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Lann Martin <a href="https://bytecodealliance.github.io/zulip-archive/stream/217126-wasmtime/topic/Timeout.20Error.20re-entry.html#442521098">(Jun 04 2024 at 12:44)</a>:</h4>
<p>Are you able to reproduce this issue on the latest wasmtime release?</p>



<a name="442521198"></a>
<h4><a href="https://bytecodealliance.zulipchat.com#narrow/stream/217126-wasmtime/topic/Timeout%20Error%20re-entry/near/442521198" class="zl"><img src="https://bytecodealliance.github.io/zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Lann Martin <a href="https://bytecodealliance.github.io/zulip-archive/stream/217126-wasmtime/topic/Timeout.20Error.20re-entry.html#442521198">(Jun 04 2024 at 12:44)</a>:</h4>
<p>(I doubt this is a wasmtime bug, but if it is then upgrading would be the first step)</p>



<a name="442521279"></a>
<h4><a href="https://bytecodealliance.zulipchat.com#narrow/stream/217126-wasmtime/topic/Timeout%20Error%20re-entry/near/442521279" class="zl"><img src="https://bytecodealliance.github.io/zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Ashanti Mutinta <a href="https://bytecodealliance.github.io/zulip-archive/stream/217126-wasmtime/topic/Timeout.20Error.20re-entry.html#442521279">(Jun 04 2024 at 12:45)</a>:</h4>
<p>I see, awesome. I will see if this goes away if I just use another error type. We are a bit gated to this version, but I think I will convince them to move. Need all these nice new features anyway.</p>



<a name="442522300"></a>
<h4><a href="https://bytecodealliance.zulipchat.com#narrow/stream/217126-wasmtime/topic/Timeout%20Error%20re-entry/near/442522300" class="zl"><img src="https://bytecodealliance.github.io/zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Lann Martin <a href="https://bytecodealliance.github.io/zulip-archive/stream/217126-wasmtime/topic/Timeout.20Error.20re-entry.html#442522300">(Jun 04 2024 at 12:50)</a>:</h4>
<blockquote>
<p>I have a set a callback that yields until it reaches a max or final value. If this value is reached it returns a trap::interrupt error.</p>
</blockquote>
<p>Another thought: are you sure this value is being reset for each instance? If it is tracked as part of the <code>Store&lt;T&gt;</code> data (<code>T</code>) and the <code>Store</code> is reused between instances that value wouldn't automatically be reset for a new instance</p>



<a name="442522893"></a>
<h4><a href="https://bytecodealliance.zulipchat.com#narrow/stream/217126-wasmtime/topic/Timeout%20Error%20re-entry/near/442522893" class="zl"><img src="https://bytecodealliance.github.io/zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Ashanti Mutinta <a href="https://bytecodealliance.github.io/zulip-archive/stream/217126-wasmtime/topic/Timeout.20Error.20re-entry.html#442522893">(Jun 04 2024 at 12:53)</a>:</h4>
<p>I think I create a store for each new instance but I should investigate if there is any reuse happening somewhere</p>



<hr><p>Last updated: Jan 20 2025 at 06:04 UTC</p>
</html>